dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v36/V2_36_23__Add_data_sharing_to_sqlview.java

```diff
@@ -0,0 +1,83 @@
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.db.migration.v36;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.hisp.dhis.user.sharing.Sharing;
+import org.hisp.dhis.util.SharingUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+/**
+ * @author Jan Bernitt
+ */
+public class V2_36_23__Add_data_sharing_to_sqlview extends BaseJavaMigration
+{
+
+    private static final Logger log = LoggerFactory.getLogger( V2_36_23__Add_data_sharing_to_sqlview.class );
+
+    @Override
+    public void migrate( Context context )
+        throws Exception
+    {
+        try (Statement statement = context.getConnection().createStatement())
+        {
+            ResultSet results = statement.executeQuery( "select sqlviewid, sharing from sqlview;" );
+            while ( results.next() )
+            {
+                updateRow( context, results.getLong( 1 ), results.getString( 2 ) );
+            }
+        }
+    }
+
+    private void updateRow( Context context, long sqlviewid, String sharing )
+        throws SQLException,
+        JsonProcessingException
+    {
+        String updatedSharing = SharingUtils.withAccess( sharing, Sharing::copyMetadataToData );
+        String update = String.format( "update sqlview set sharing = '%s'::json where sqlviewid = %d", updatedSharing,
```

Comment:
We could potentially use preparedStatement instead of statement? It may not be a big difference, unless there is significantly huge number of sql views in an instance, which is highly unlikely. What do you think?

Reply:
SQL views are unlikely to be many but since this might be used as template for other tables where we add data sharing I'll use prepared statements instead.